### PR TITLE
VZ-4869 restore url

### PR DESCRIPTION
**Pull request by theabmitra**
# Description

Today at install vz instance is updated with conditions and install urls under .status.instance . On a restore of VZ instance, these dont get populated. Since the components are already installed, with this change VPO will update them . 
This is required since console fetches them and displays in UI.  

Fixes VZ-9999

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by kminder — COMMENTED** [review]
These changes look ok to me but need to reviewed by someone more familiar with VPO.  I suggest Paul Mackin and Mike Cico.

In addition there needs to be a test added that covers this change specifically.

File: platform-operator/controllers/verrazzano/upgrade_test.go [hunk 1]
```diff
@@ -84,6 +85,21 @@ func TestUpgradeNoVersion(t *testing.T) {
...
+		List(gomock.Any(), gomock.Not(gomock.Nil())).
+		DoAndReturn(func(ctx context.Context, ingressList *networkingv1.IngressList) error {
+			ingressList.Items = []networkingv1.Ingress{}
+			return nil
+		}).AnyTimes()
+	mock.EXPECT().Status().Return(mockStatus).AnyTimes()
+	mockStatus.EXPECT().
+		Update(gomock.Any(), gomock.Any()).
+		DoAndReturn(func(ctx context.Context, verrazzano *vzapi.Verrazzano, opts ...client.UpdateOption) error {
+			asserts.NotZero(len(verrazzano.Status.Components), "Status.Components len should not be zero")
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/upgrade_test.go:99]: Do you need to validate that the URLs are present?  So it looks like this is verifying the `status.components` field is not empty, which is good.  

I'm wondering if it would be better to break out this specific test case into its own unit test, instead of side-effect testing here.

If you do, you could minimize or avoid using the mocks possibly by using the `fakeComponent` struct and overriding the `GetComponents` func.  There are other places where we do that, like this

```
	registry.OverrideGetComponentsFn(func() []spi.Component {
		return []spi.Component{
			fakeComponent{},
		}
	})
	defer registry.ResetGetComponentsFn()
```

Comment (theabmitra) [reply platform-operator/controllers/verrazzano/upgrade_test.go:99]: Well we will still need the expect in the tests, as the List, Status and Update will be called always when Component status is Ready. 
I can add a non zero check for ingress list as well.

Comment (kminder) [reply platform-operator/controllers/verrazzano/upgrade_test.go:175]: I was also expecting to see some test where the URLs were initially empty and after a call to reconcileComponents they were corrected.

Comment (theabmitra) [reply platform-operator/controllers/verrazzano/upgrade_test.go:111]: @kminder and @michael-cico added new tests with the scenarios. please check

**Review by michael-cico — APPROVED** [review]

**Review by kminder — APPROVED** [review]
Approved.

File: platform-operator/controllers/verrazzano/upgrade_test.go [hunk 2]
```diff
@@ -1503,3 +1545,410 @@ func (r goodRunner) Run(_ *exec.Cmd) (stdout []byte, stderr []byte, err error) {
...
+	result, err := reconciler.Reconcile(request)
+
+	// Validate the results
+	mocker.Finish()
+	asserts.NoError(err)
+	asserts.Equal(false, result.Requeue)
+	asserts.Equal(time.Duration(0), result.RequeueAfter)
+
+	// validating instance urls are updated
+	fakeInstanceInfo := vzapi.InstanceInfo{}
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/upgrade_test.go:1905]: This should be done in the update status mock

Comment (theabmitra) [reply platform-operator/controllers/verrazzano/upgrade_test.go:1905]: This was added to cover the case under normal operation, where instance info is already populated.
